LCJava/src/main/java/tw/TrainRoute.java

```diff
@@ -0,0 +1,195 @@
+package tw;
+
+import java.io.BufferedReader;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.*;
+
+public class TrainRoute {
+
+    private class ResultMessage {
+        public static final String SUCCESS =  "SUCCESS";
+        public static final String FAILURE = "NO SUCH ROUTE";
+    }
+
+    private Map<String, Map<String, Integer>> graph;
+
+
+    public TrainRoute (String filename) {
+        this.graph = new HashMap<>();
+        this.initGraph(filename);
+    }
+
+    private String readGraphSource(String filename) throws IOException, FileNotFoundException {
+        /*
+        if (filename.length() > 0) {
+            return "AB5,BC4,CD8,DC8,DE6,AD5,CE2,EB3,AE7";
+        }
+        */
+        BufferedReader dataFile = new BufferedReader(new FileReader(filename));
+        String line, data;
+        data = "";
+        while ((line = dataFile.readLine()) != null) {
+            data += line;
+        }
+        return data;
+    }
+
+    private void initGraph(String file) {
+        String graphRaw = "";
+        try {
+            graphRaw = readGraphSource(file);
+        } catch (Exception e) {
+            System.out.println("Couldn't read raw graph data");
+            System.exit(-1);
+        }
+        try {
+            parseGraph(graphRaw);
+        } catch (TrainMapException e) {
+            System.out.println(e.errorMsg());
+            System.out.println("Couldn't parse raw graph data");
+            System.exit(-1);
+        }
+
+
+    }
+    private void parseGraph(String rawGraph) throws TrainMapException {
+        String[] input = rawGraph.split(",");
+
+        for (int i = 0; i < input.length; i++) {
+            //TODO first, second must be Character
+            if (!Character.isLetter(input[i].charAt(0)) || !Character.isLetter(input[i].charAt(1))) {
+                throw new TrainMapException("Invalid Raw Graph Data: Town must be letter");
+            }
+            String startTown = input[i].substring(0, 1);
+            String endTown = input[i].substring(1, 2);
+            int distance = 0;
+            try {
+                distance = Integer.parseInt(input[i].substring(2));
+            } catch (Exception e) {
```

Comment:
catch the specific exception that parseInt may throw.

do 
```
catch (NumberFormatException e)
```
it's an anti pattern to use `Exception` to catch all sorts of exception without a good reason.